protocols/smtp/src/main/java/org/apache/james/protocols/smtp/core/DataCmdHandler.java

```diff
@@ -126,13 +137,19 @@ public void destroy() {
      *
      */
     public Response onCommand(SMTPSession session, Request request) {
-        String parameters = request.getArgument();
-        Response response = doDATAFilter(session,parameters);
-        
-        if (response == null) {
-            return doDATA(session, parameters);
-        } else {
-            return response;
+        TimeMetric timeMetric = metricFactory.timer("SMTP-" + request.getCommand());

```

Comment:
Don't you think it would be great to use a try with statement (https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html) by making TimeMetric autocloseable ?